TNT: Raise cost of TNT by adding a TNT stick crafting stage

paramat opened the pull request:
TNT: Raise cost of TNT by adding a TNT stick crafting stage
6 gunpowder and 1 paper crafts to 2 TNT stick craftitems.
9 TNT sticks craft to 1 TNT.
TNT stick is not yet usable as an explosive, possibly later.
////////////////

![stick_diagonal_dark128](https://user-images.githubusercontent.com/3686677/40815446-7999b6f8-653e-11e8-879c-571df4e9dca3.png)

For #1482 
Remove wood from TNT recipe since it is a bundle of TNT sticks, use paper instead.
Stick texture derived from benrob0329's texture below, with these changes:
* Pixels use hue and saturation matching the TNT texture, 3 of the pixel colours are identical to the TNT texture pixels.
* Stick is slightly longer to be closer to the slimness of the sticks in the TNT texture.

sfan5 commented on mods/tnt/README.txt:
```diff
@@ -23,18 +23,21 @@ All gunpowder textures except tnt_gunpowder_inventory.png.
--- wood - gunpowder -- wood -
-gunpowder gunpowder gunpowder
--- wood - gunpowder -- wood -
+The gunpowder can be used to craft TNT sticks or as fuse for TNT.
```
s/as/a/

paramat replied:
Updated.

Fixer-007 commented:
Not bad if you make it explosive throw-able alongside with snow and others via throwing api =)

paramat commented:
It seems fairly obvious to make this a node of low explosive power but i'm not sure how usable 1 node of destruction is.

random-geek commented:
I would suggest adding paper to the crafting recipe for the TNT sticks to hold them together. Adding one paper to the TNT stick crafting recipe, however, would require an additional 27 papyrus per stick. Therefore, my idea is to make 2 sticks craftable from 5 gunpowder and 1 paper. This would make the cost of one TNT block equal 4.5 each gravel and coal (still more than the 1 each which is required now) and 13.5 papyrus, the latter being easily obtainable with a farm and a diamond sword.

paramat commented:
README updated @sfan5 

I considered adding paper to the recipe but decided against since papyrus would need to be farmed. Also the problem of the amount of paper needed, and the partial workarounds for that which make crafting recipes less intuitive.

I looked at the tnt API, it can't be used for a nodebox TNT stick, so if a stick is made usable as an explosive it would be extra dedicated code. I'm not sure it's worth although players will probably expect it.

Fixer-007 commented:
>It seems fairly obvious to make this a node of low explosive power but i'm not sure how usable 1 node of destruction is.

To use it as a throw-able grenade in PVP or to destroy just few blocks...

rubenwardy commented:
Add it as a craft item now, when we add a throwing mod it should make it throwable

Napiophelios commented:
I like the idea of a single stick of TNT, but raising the cost of coal and gravel 
while removing the wood components just makes this easy game even easier. 

I like the idea of having to find or farm the papyrus for the dynamite.
Being able to craft this without having to return to the surface for components tho is just too easy.

just my $00.02  :)

Ezhh commented:
I'm fairly neutral on this, but adding a stick item that currently does nothing seems weird to me, and this only seems to make the crafting of TNT more fiddly/annoying rather than more expensive.  The ingredients also remain really common ones and, unlike before, you can now find all of them in the same place.

If the goal is to eventually turn the sticks into smaller scale but throwable explosives, this will actually have the end result of much cheaper individual explosives as well.

random-geek commented:
Much cheaper explosives, but would only destroy a small fraction of what regular TNT could. I like the idea of TNT sticks, as long as they aren't _completely_ pointless.

As far as not having to return to the surface for supplies, I don't understand the logic behind this. We already have to get sticks for pickaxes, shovels, and torches. By the nature of the game, players must use materials from both above and below the ground to sustain themselves and progress.

What if we changed the TNT craftrecipe to just 1 paper and 8 TNT sticks? It wouldn't take too much time to gather the papyrus, and would still require _some_ above-ground material.

paramat commented:
Good points, the ease of crafting has not necessarily ncreased.
Perhaps i can make 2 sticks return from:
G_G
GPG
G_G
G gunpowder
P paper
?
Krock also likes the paper idea.

Ezhh commented:
TNT should be something you progress toward being able to produce in any real quantity. It shouldn't be something you make right away from more-or-less the start of the game.

> I like the idea of TNT sticks, as long as they aren't completely pointless.

Same thought, and with the current recipes it feels like individual sticks would need to be underpowered, especially if throwable, or they'd be too cheap for what they are. I'd rather they take some effort and be worthwhile. Of course, they can't ever be too powerful, or people won't bother to turn them into TNT nodes...

Either way, I'd be happy to see paper needed. It would also give paper more use, which isn't a bad thing.

paramat commented:
Ok will make the suggested change and add paper.

numberZero commented:
I suggest using 1 or 2 paper per stick; this way, recipes are better than if crafting 2 sticks at once.

Besides, while the TNT node is textured currently as a bunch of sticks, that doesn’t mean that’s the only way; take a look at Vanessa’s texture for example. So it may make sense to use some wood in the recipe (or even a chest?..).

paramat commented:
We don't worry about alternative textures as those could be anything, MTG is coded for the nodes as they appear in default textures. Texture packs are expected to depict the same item.
However saying that it is of course silly how we are crafting gunpowder into TNT which is a completely different explosive, but too late.

numberZero commented:
@paramat Default textures can be changed, if necessary.

numberZero commented:
BTW, I like your TNT stick texture)

Ferk commented:
The stick should probably have ``flammable = 5`` like the TNT.

That makes me think that maybe the check that already exists for the items that fall in fire/lava could be updated to trigger the ``on_ignite`` when the dropped item/node has one, instead of being destroyed.

That way a dropped TNT node or stick (if later on a lower impact on_ignite was added to it) would explode if someone drops it into lava/fire rather than making a smoke puff.

paramat commented:
numberZero yes but it seems a dilemma: the mod is called TNT and we have TNT items in worlds, but also gunpowder items essential for fuse trails. Whether we depict the explosive a s a crate of gunpowder or TNT we have inconsistency.
It seems probably best to stay with how the item has been depicted for years in default textures (and therefore also texture packs): as a bundle of TNT sticks, therefore adding a single stick and using paper in the recipe seems best to match the texture.

paramat commented:
Updated with 'flammable' and paper in the recipe.

paramat commented:
Added a 2nd commit.
To continue our effort in minimising the effect of inherently intensive ABMs i have increased the interval of the ignition ABM from 4 to 9s. Still seems a reasonable time for a nearby heat source to ignite gunpowder or TNT.
9 was chosen as it is the lowest unused time in the list of MTG ABM intervals. We deliberately give each ABM a unique interval in an effort to reduce the number of ABMs that trigger simultaneously, therefore spreading the load.
```
MTG ABMs:

default:
lavacool 2
cactus 12
papyrus 14
grass spread 6
grass death 8
moss 16
coral death 17

farming:
soil 15

tnt:
ignite 4

fire:
extinguish flame 3
remove disabled fire 7 (usually disabled)
ignite flame 7
remove flammable nodes 5

flowers:
flower spread 13
mushroom spread 11

stairs:
slab replace 16 (usually disabled)

14 ABMs running by default.

2
3
4
5
6
7
(7)
8
-
-
11
12
13
14
15
16
(16)
17
```

paramat commented:
@SmallJoker

benrob0329 commented:
Made a new texture for it :wink: 
![tnt_stick](https://user-images.githubusercontent.com/19180682/40746014-1eaac57e-6427-11e8-9b2d-c51225a8b51a.png)
![tnt_stick_scaled](https://user-images.githubusercontent.com/19180682/40746064-516fe296-6427-11e8-9039-5c0c311f4f40.png)

SmallJoker commented on mods/tnt/README.txt:
```diff
@@ -23,22 +23,33 @@ All gunpowder textures except tnt_gunpowder_inventory.png.
   3. Activate it with mesecons (fastest way).
 
-Be aware of the damage radius of 6 blocks!
+Be aware of the player damage radius of 6 blocks!
```
Not limited to players. All mortal objects will take damage, unless on_blast is defined.
`blocks` -> `nodes` (or metres)

paramat replied:
Ok.

SmallJoker commented:
@benrob0329 I like the shading of this texture and would like to see it used in this PR. However, the end of the ignition wire is too close to the main part.

benrob0329 commented:
![tnt_stick_scaled](https://user-images.githubusercontent.com/19180682/40790738-ab384e3e-64c3-11e8-92d2-19545254c52c.png)

![tnt_stick](https://user-images.githubusercontent.com/19180682/40790676-840889dc-64c3-11e8-9320-482093337f79.png)

paramat commented:
Updated.
Removed 2nd commit.
I did consider using a diagonal stick texture, so have derived a new texture from benrob0329's texture.
See first post.

paramat commented:
@Ezhh are you ok with this being merged?

Ezhh commented:
No issues with concept, but I have no time to test things right now.
